src/widgets/dialogs/UserInfoPopup.cpp

```diff
@@ -536,25 +554,19 @@ UserInfoPopup::TimeoutWidget::TimeoutWidget()
 
     addButton(Unban, "unban", getApp()->resources->buttons.unban);
 
-    addTimeouts("sec", {{"1", 1}});
-    addTimeouts("min", {
-                           {"1", 1 * 60},
-                           {"5", 5 * 60},
-                           {"10", 10 * 60},
-                       });
-    addTimeouts("hour", {
-                            {"1", 1 * 60 * 60},
-                            {"4", 4 * 60 * 60},
-                        });
-    addTimeouts("days", {
-                            {"1", 1 * 60 * 60 * 24},
-                            {"3", 3 * 60 * 60 * 24},
-                        });
-    addTimeouts("weeks", {
-                             {"1", 1 * 60 * 60 * 24 * 7},
-                             {"2", 2 * 60 * 60 * 24 * 7},
-                         });
+    const auto timeoutButtons = getSettings()->timeoutButtons.getValue();
+    std::vector<std::pair<QString, int>> t(8);  // Timeouts.

```

Comment:
Oh, I missed that `std::pair<QString, int>` could be replaced with `TimeoutButton`.